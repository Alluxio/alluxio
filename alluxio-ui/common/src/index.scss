/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

@import './constants/scss/themeVariables';
@import '~bootstrap/scss/bootstrap';
@import '~bootstrap/scss/bootstrap-grid';
@import '~bootstrap/scss/bootstrap-reboot';
@import './constants/scss/themeCustomization';
@import '~source-sans-pro/source-sans-pro.css';

body {
  margin: 0;
  padding: 0;
  font-family: 'Source Sans Pro', sans-serif;
  font-size: 1em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-shadow: rgba(0,0,0,.01) 0 0 1px;
  text-shadow: rgba(0,0,0,.01) 0 0 1px;
}
