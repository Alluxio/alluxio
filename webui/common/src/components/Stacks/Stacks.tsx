/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import { AnyAction, compose, Dispatch } from 'redux';

import { withErrors, withFetchData, withLoadingMessage } from '../';
import { ICommonState, IRequest } from '../../constants';
import { fetchRequest } from '../../store/stacks/actions';
import { Form, FormGroup, Input } from 'reactstrap';
import { History, LocationState } from 'history';

export interface IStackPropsFromState extends ICommonState {
  stackData: string;
}

interface IPropsFromDispatch {
  fetchRequest: typeof fetchRequest;
}

interface IStacksProps {
  history: History<LocationState>;
  location: { search: string };
  queryStringSuffix: string;
  request: IRequest;
}

export type AllStacksProps = IStackPropsFromState & IPropsFromDispatch & IStacksProps;

export class Stacks extends React.Component<AllStacksProps> {
  public render(): JSX.Element {
    const { stackData } = this.props;
    return (
      <div className="col-12">
        <Form className="mb-3 viewData-file-form" id="viewDataFileForm" inline={true}>
          <FormGroup className="mb-2 mr-sm-2 w-100">
            <Input className="w-100" type="textarea" value={stackData} style={{ height: '80vh' }} readOnly={true} />
          </FormGroup>
        </Form>
      </div>
    );
  }
}

export const mapDispatchToStacksProps = (dispatch: Dispatch): { fetchRequest: () => AnyAction } => ({
  fetchRequest: (): AnyAction => dispatch(fetchRequest()),
});

export default compose(
  withFetchData,
  withErrors,
  withLoadingMessage,
)(Stacks) as typeof React.Component;
