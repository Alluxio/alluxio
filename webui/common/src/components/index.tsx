/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export * from './HOCs/withLoadingMessage/withLoadingMessage';
export * from './HOCs/withErrors/withErrors';
export * from './HOCs/withTextAreaResize/withTextAreaResize';
export * from './HOCs/withFetchDataFromPath/withFetchDataFromPath';
export * from './HOCs/withFetchData/withFetchData';
export * from './HOCs/withFluidContainer/withFluidContainer';
export * from './Footer/Footer';
export * from './Header/Header';
export * from './LoadingMessage/LoadingMessage';
export * from './Logs/Logs';
export * from './FileView/FileView';
export * from './Paginator/Paginator';
export * from './LineGraph/LineGraph';
