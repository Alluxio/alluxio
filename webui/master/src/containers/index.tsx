/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export {default as App} from './App/App';

export {default as Browse} from './pages/Browse/Browse';
export {default as Configuration} from './pages/Configuration/Configuration';
export {default as Data} from './pages/Data/Data';
export {default as Overview} from './pages/Overview/Overview';
export {default as MasterLogs} from './pages/MasterLogs/MasterLogs';
export {default as Workers} from './pages/Workers/Workers';
export {default as Metrics} from './pages/Metrics/Metrics';
