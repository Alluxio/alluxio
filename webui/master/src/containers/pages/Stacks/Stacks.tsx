/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';

export class Stacks extends React.Component {
  public render(): JSX.Element {
    const path = 'http://' + location.host + location.pathname;
    return (
      <div className="stacks-page" style={{ backgroundColor: 'white', minHeight: window.innerHeight }}>
        <iframe style={{ width: '100%', height: window.innerHeight, overflow: 'visible' }} src={path} />
      </div>
    );
  }
}
export default Stacks as typeof React.Component;
