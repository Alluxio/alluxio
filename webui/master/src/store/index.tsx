/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {browseReducer, initialBrowseState} from './browse/reducer';
import {browseSaga} from './browse/sagas';
import {IBrowseState} from './browse/types';
import {configReducer, initialConfigState} from './config/reducer';
import {configSaga} from './config/sagas';
import {IConfigState} from './config/types';
import {dataReducer, initialDataState} from './data/reducer';
import {dataSaga} from './data/sagas';
import {IDataState} from './data/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';
import {IInitState} from './init/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {initialWorkersState, workersReducer} from './workers/reducer';
import {workersSaga} from './workers/sagas';
import {IWorkersState} from './workers/types';

export interface IApplicationState {
  browse: IBrowseState;
  config: IConfigState;
  data: IDataState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
  workers: IWorkersState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  browse: browseReducer,
  config: configReducer,
  data: dataReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history),
  workers: workersReducer
});

export const rootSaga = function* () {
  yield all([
    fork(browseSaga),
    fork(configSaga),
    fork(dataSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga),
    fork(workersSaga)
  ]);
};

export const initialState: IApplicationState = {
  browse: initialBrowseState,
  config: initialConfigState,
  data: initialDataState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState,
  workers: initialWorkersState
};
