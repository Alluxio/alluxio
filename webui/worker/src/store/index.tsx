/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {blockInfoReducer, initialBlockInfoState} from './blockInfo/reducer';
import {blockInfoSaga} from './blockInfo/sagas';
import {IBlockInfoState} from './blockInfo/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {IInitState} from './init/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';

export interface IApplicationState {
  blockInfo: IBlockInfoState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  blockInfo: blockInfoReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history)
});

export const rootSaga = function* () {
  yield all([
    fork(blockInfoSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga)
  ]);
};

export const initialState: IApplicationState = {
  blockInfo: initialBlockInfoState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState
};
