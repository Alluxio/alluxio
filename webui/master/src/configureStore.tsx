/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {routerMiddleware} from 'connected-react-router';
import {History} from 'history';
import {applyMiddleware, createStore, Store} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import {IApplicationState, rootReducer, rootSaga} from './store';

export default function configureStore(history: History, initialState: IApplicationState): Store<IApplicationState> {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();

  const store = createStore(
    rootReducer(history),
    initialState,
    composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware))
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
