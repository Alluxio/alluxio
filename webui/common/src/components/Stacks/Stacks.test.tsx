/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createBrowserHistory, History, LocationState } from 'history';
import React from 'react';
import sinon from 'sinon';
import { AllStacksProps, Stacks } from './Stacks';
import { initialState } from '../../../../master/src/store';
import { createAlertErrors } from '../../utilities';

configure({ adapter: new Adapter() });

describe('Stacks', () => {
  let history: History<LocationState>;
  let props: AllStacksProps;

  beforeAll(() => {
    history = createBrowserHistory({ keyLength: 0 });
    history.push('/stacks');
    props = {
      location: { search: '' },
      history: history,
      fetchRequest: sinon.spy(() => {}),
      stackData: initialState.stacks.data,
      refresh: initialState.refresh.data,
      queryStringSuffix: '',
      request: {},
      loading: false,
      class: '',
      errors: createAlertErrors(false),
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<Stacks {...props} />);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Contains a div with class col-12', () => {
      expect(shallowWrapper.find('.col-12').length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
