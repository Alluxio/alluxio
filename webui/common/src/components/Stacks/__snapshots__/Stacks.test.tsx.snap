// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stacks Shallow component Matches snapshot 1`] = `
<div
  className="col-12"
>
  <Form
    className="mb-3 viewData-file-form"
    id="viewDataFileForm"
    inline={true}
    tag="form"
  >
    <FormGroup
      className="mb-2 mr-sm-2 w-100"
      tag="div"
    >
      <Input
        className="w-100"
        readOnly={true}
        style={
          Object {
            "height": "80vh",
          }
        }
        type="textarea"
        value=""
      />
    </FormGroup>
  </Form>
</div>
`;
